### PR TITLE
error_messages

### PR DESCRIPTION
**Pull request by negarn**

### CODE REVIEW

File: src/javascript/app_2/App/Containers/ServicesErrorModal/services-error-modal.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React         from 'react';
+import PropTypes     from 'prop-types';
+import { localize }  from '_common/localize';
+import FullPageModal from 'App/Components/Elements/FullPageModal/full-page-modal.jsx';
+import { connect }   from 'Stores/connect';
+import { title }    from './constants';
```
Comment (zakibinary) [inline src/javascript/app_2/App/Containers/ServicesErrorModal/services-error-modal.jsx]: Misalignment here

File: src/javascript/app_2/Modules/Trading/Containers/purchase.jsx [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 import { localize }      from '_common/localize';
-import { isEmptyObject } from '_common/utility';
 import { PopConfirm }    from 'App/Components/Elements/PopConfirm';
+import Tooltip           from 'App/Components/Elements/tooltip.jsx';
 import Fieldset          from 'App/Components/Form/fieldset.jsx';
 import { connect }       from 'Stores/connect';
 import ContractInfo      from '../Components/Form/Purchase/contract-info.jsx';
-import MessageBox        from '../Components/Form/Purchase/MessageBox';
 import PurchaseLock      from '../Components/Form/Purchase/PurchaseLock';
 import PurchaseButton    from '../Components/Elements/purchase-button.jsx';
```
Comment (zakibinary) [inline src/javascript/app_2/Modules/Trading/Containers/purchase.jsx]: extra whitespace after `import { isEmptyObject }` removed.

Comment (negarn) [reply src/javascript/app_2/Modules/Trading/Containers/purchase.jsx]: I don't think so, that's just how github displays it. there's no line number, it's empty.

File: src/javascript/app_2/App/Components/Form/InputField/input-field.jsx [hunk 3]
```diff
@@ -186,7 +186,7 @@ const InputField = ({
         />;
 
     const input_tooltip =
-        <Tooltip className={classNames('', { 'with-label': label })} alignment='left' message={has_error ? error_messages[0] : null } has_error={has_error}>
+        <Tooltip className={classNames('', { 'tooltip--with-label': label })} alignment='left' message={has_error ? error_messages[0] : null } has_error={has_error}>
```
Comment (zakibinary) [inline src/javascript/app_2/App/Components/Form/InputField/input-field.jsx]: Can we break down the props into separate lines?

File: src/javascript/app_2/App/Containers/ServicesErrorModal/services-error-modal.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            // onCancel={onCancel}
+            is_visible={is_services_error_visible}
+        >
+            {message}
+        </FullPageModal>
+    );
+};
+
+ServicesErrorModal.propTypes = {
+    services_error: PropTypes.object,
```
Comment (OskarAhl) [inline src/javascript/app_2/App/Containers/ServicesErrorModal/services-error-modal.jsx]: `is_services_error_visible ` and `toggleServicesErrorModal` propTypes should be here

File: src/javascript/app_2/App/Components/Form/InputField/input-field.jsx [hunk 5]
```diff
@@ -186,7 +186,12 @@ const InputField = ({
         />;
 
     const input_tooltip =
-        <Tooltip className={classNames('', { 'with-label': label })} alignment='left' message={has_error ? error_messages[0] : null } has_error={has_error}>
+        <Tooltip
+            className={classNames('', { 'tooltip--with-label': label })}
+            alignment='left'
+            message={has_error ? error_messages[0] : null }
```
Comment (cakasuma) [inline src/javascript/app_2/App/Components/Form/InputField/input-field.jsx]: nit extra space after null

File: src/javascript/app_2/Modules/Trading/Containers/purchase.jsx [hunk 6]
```diff
@@ -57,24 +60,22 @@ const Purchase = ({
...
+                        basis={basis}
                         currency={currency}
                         proposal_info={info}
                         has_increased={info.has_increased}
+                        is_loading={is_loading}
                         is_visible={!is_contract_mode}
                     />
-                    <div className='btn-purchase__shadow-wrapper'>
+                    <div className={classNames('btn-purchase__shadow-wrapper', { 'btn-purchase__shadow-wrapper--disabled': (is_proposal_error || is_disabled) })}>
+                        {is_proposal_error &&
```
Comment (cakasuma) [inline src/javascript/app_2/Modules/Trading/Containers/purchase.jsx]: space and align tooltip ?

Comment (negarn) [reply src/javascript/app_2/Modules/Trading/Containers/purchase.jsx]: it's following the same style as line 67-69

**Pull request merged**